index.js

```diff
@@ -2,38 +2,123 @@ const readline = require('readline');
 const readlineInterface = readline.createInterface(process.stdin, process.stdout);
 
 function ask(questionText) {
-  return new Promise((resolve, reject) => {
-    readlineInterface.question(questionText, resolve);
-  });
+    return new Promise((resolve, reject) => {
+        readlineInterface.question(questionText, resolve);
+    });
 }
-
-// remember the StateMachine lecture
-// https://bootcamp.burlingtoncodeacademy.com/lessons/cs/state-machines
+/////////state machine
 let states = {
-  'roomOne': { canChangeTo: [ 'roomTwo' ] },
-  'roomTwo': { canChangeTo: [ 'roomThree' ] },
-  'roomThree': { canChangeTo: [ 'roomOne' ] }
+    '182main': { canChangeTo: ['foyer', 'muddy waters', 'nectars'] },
+    'foyer': { canChangeTo: ['182main', 'classroom'] },
+    'classroom': { canChangeTo: ['foyer'] },
+    'muddy waters': { canChangeTo: ['182main'] }
 };
 
-let currentState = "green";
-
 function enterState(newState) {
-  let validTransitions = states[currentState].canChangeTo;
-  if (validTransitions.includes(newState)) {
-    currentState = newState;
-  } else {
-    throw 'Invalid state transition attempted - from ' + currentState + ' to ' + newState;
-  }
+    let validTransitions = states[currentState].canChangeTo;
+    if (validTransitions.includes(newState)) {
+        currentState = newState;
+    } else {
+        console.log("Sorry, you can't go there directly from here.")
+        return this;
+    }
 }
+////////end state machine
 
+let currentState = "182main";
 start();
 
 async function start() {
-  const welcomeMessage = `182 Main St.
+    const welcomeMessage = `182 Main St.
 You are standing on Main Street between Church and South Winooski.
 There is a door here. A keypad sits on the handle.
-On the door is a handwritten sign.`;
-  let answer = await ask(welcomeMessage);
-  console.log('Now write your code to make this work!');
-  process.exit();
+On the door is a handwritten sign.\n`;
+    let answer = await ask(welcomeMessage);
+    for (; answer;) {
+        if (answer.toLowerCase() === "open door") {
+            answer = await ask("The door is locked. There is a keypad on the door handle.\n");
+        }
+        else if (answer.toLowerCase() === "take sign")
+            answer = await ask("That would be selfish. How will other students find their way?\n");
+        else if (answer.toLowerCase() !== "read sign") {
+            answer = await ask('Sorry, I don\'t understand "' + answer + '."\n');
+        }
+        else {
+            sign();
+            break;
+        }
```

Comment:
This logic looks a little backwards to me, but it does work. In general it's better practice to use your `else` as your "catch all" which you could do if you're checking `if (answer.toLowerCase() === "read sign") in the preceding logic check.